tools/ideabuck/src/com/facebook/buck/intellij/ideabuck/run/BuckRunConfigurationExtension.java

```diff
@@ -0,0 +1,171 @@
+import com.facebook.buck.intellij.ideabuck.build.ResultCallbackBuckHandler;
+import com.facebook.buck.intellij.ideabuck.config.BuckModule;
+import com.facebook.buck.intellij.ideabuck.file.BuckFileUtil;
+import com.google.common.base.Function;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.intellij.execution.BeforeRunTask;
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.RunConfigurationExtension;
+import com.intellij.execution.RunManager;
+import com.intellij.execution.application.ApplicationConfiguration;
+import com.intellij.execution.configurations.JavaParameters;
+import com.intellij.execution.configurations.RunConfigurationBase;
+import com.intellij.execution.configurations.RunnerSettings;
+import com.intellij.execution.impl.RunManagerImpl;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiClass;
+import com.intellij.util.PathsList;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.jetbrains.annotations.NotNull;
+
+public class BuckRunConfigurationExtension extends RunConfigurationExtension {
+
+  private String targetResult;
+  private List<String> additionalClasspath;
+  private Project project;
+
+  @Override
+  public <T extends RunConfigurationBase> void updateJavaParameters(T configuration,
+      JavaParameters javaParameters, RunnerSettings runnerSettings) throws ExecutionException {
+    Project currentProject = configuration.getProject();
+    VirtualFile containingFile;
+    VirtualFile buckFile;
+    if (configuration instanceof ApplicationConfiguration) {
+      containingFile = ((ApplicationConfiguration) configuration).getMainClass()
+          .getContainingFile().getVirtualFile();
+      buckFile = BuckFileUtil.getBuckFile(containingFile);
+    } else {
+      return;
+    }
+
+    BuckBuildManager buildManager = BuckBuildManager.getInstance(currentProject);
+    BuckQueryCommandHandler buckQueryCommandHandler =
+        new BuckQueryCommandHandler(
+            currentProject,
+            buckFile.getParent(),
+            BuckCommand.QUERY,
+            new Function<List<String>, Void>() {
+              @javax.annotation.Nullable
+              @Override
+              public Void apply(@javax.annotation.Nullable List<String> strings) {
+                if (strings == null
+                    || strings.isEmpty()
+                    || strings.get(0) == null
+                    || strings.get(0).isEmpty()) {
+                  return null;
+                }
+                String targetResultCandidate = strings.get(0);
+                for (String candidate : strings) {
+                  if (!candidate.contains("lint")) {
```

Comment:
looks like duplicated logic.

Reply:
Hard not to duplicate here since it has to pass the target results to other buck commands.